### ISSUE TITLE
Hammer seems does not work in Phonegap project!

### ISSUE THREAD

**Issue by tthanhlong** [hammerjs/hammer.js#758]
That sounds great, this JS plugin. But I met an issue with Hammer and need your help.

I am developing phonegap with Monaca tool. I added hammer.js and created very simple:
        var myElement = document.getElementById('myElement');

        // create a simple instance
        // by default, it only adds horizontal recognizers
        var mc = new Hammer(myElement);
        alert("as");
        // listen to events...
        mc.on("panleft panright tap press", function(ev) {
            myElement.textContent = ev.type +" gesture detected.";
        });

Unluckily, on device there are no effect, no error found. When testing on browser, "Uncaught ReferenceError: Hammer is not defined" error is showed.
Did I miss any? Please help.

Thanks for your support.

**Comment by runspired** [comment]
Hammer is an AMD module that does not get exposed as a global in cordova/phonegap.  If you are building in an old globals structure you will need to add `window[exportName] = Hammer;` or similar to line 2462 in the non minified version.

**Comment by okwei2000** [comment]
It works fine for me.

**Comment by enrichit** [comment]
@runspired It's common practice to use AMD only when it's implemented. If you look at the bottom of hammer.js you will see this:

    if (typeof define == TYPE_FUNCTION && define.amd) {
       define(function() {
           return Hammer;
       });
    } else if (typeof module != 'undefined' && module.exports) {
       module.exports = Hammer;
    } else {
        window[exportName] = Hammer;
    }

In other words:

* If using AMD, use that.
* Else if CommonJS, use that.
* Else export to global window object.

**Comment by runspired** [comment]
@enrichit phonegap/cordova has it's own loader that's not really AMD but screws up that test, leading to it being inaccessible.

**Comment by runspired** [comment]
Hammer does work in phonegap/cordova projects, but requires using `require` or implementing global override as seen above.
